libres/lib/analysis/ies/ies_enkf.cpp

```diff
@@ -330,88 +190,182 @@ void ies_enkf_updateA(
     for (int i = 0; i < ens_size; i++) {
         matrix_iadd(X, i, i, 1.0);
     }
-    if (dbg)
-        matrix_pretty_fprint_submat(X, "X", "%11.5f", log_fp, 0, m_ens_size - 1,
-                                    0, m_ens_size);
-
-    /* COMPUTE NEW ENSEMBLE SOLUTION FOR CURRENT ITERATION  Ei=A0*X (Line 11)*/
-    matrix_pretty_fprint_submat(A, "A^f", "%11.5f", log_fp, 0, m_state_size, 0,
-                                m_ens_size);
-    ies_enkf_linalg_extract_active_A0(data, A0, log_fp);
-    matrix_matmul(A, A0, X);
-    matrix_pretty_fprint_submat(A, "A^a", "%11.5f", log_fp, 0, m_state_size, 0,
-                                m_ens_size);
 
     /* COMPUTE ||W0 - W|| AND EVALUATE COST FUNCTION FOR PREVIOUS ITERATE (Line 12)*/
     matrix_type *DW = matrix_alloc(ens_size, ens_size);
     matrix_sub(DW, W0, W);
-    teclog(W, D, DW, "iesteclog.dat", ens_size, iteration_nr, rcond, nrsing,
-           nrobs_inp);
-    teccost(W, D, "costf.dat", ens_size, iteration_nr);
 
-    ies_enkf_data_fclose_log(data);
+    std::vector<double> costJ(ens_size);
+    double costf = 0.0;
+    for (int i = 0; i < ens_size; i++) {
+        costJ[i] = matrix_column_column_dot_product(W, i, W, i) +
```

Comment:
If I am not mistaken, then the calculation of `costf` was only relevant for `tecplot`? I guess this can be removed now.

Reply:
See comments below ⬇️